sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -231,4 +232,18 @@ object ResolveDefaultColumns {
       }
     }
   }
+
+  def checkDataSourceSupportsDefaultColumns(table: CatalogTable): Unit = {
+    if (table.schema.fields.map(_.metadata).exists { m =>
+      m.contains(CURRENT_DEFAULT_COLUMN_METADATA_KEY) ||
+        m.contains(EXISTS_DEFAULT_COLUMN_METADATA_KEY)
+    }) {
+      table.provider.getOrElse("").toLowerCase() match {
+        case "csv" | "json" | "parquet" | "orc" =>
```

Comment:
Would you like to make a configuration option? Or does it depend on the data source implementation. For example, can we have DEFAULT in JDBC?

Do we also need to consider DataSource V2? Will this feature work there?

Reply:
Good ideas! I made this a configuration option. It does support DataSource V2, I added a couple more tests to show that.